sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionDescription.java

```diff
@@ -42,14 +42,17 @@
  * `group()` describes the category that the expression belongs to. The valid value is
  * "agg_funcs", "array_funcs", "datetime_funcs", "json_funcs", "map_funcs" and "window_funcs".
  *
+ * `functionType()` describe the type of the function. The valid value is "built-in", "hive",
+ * "python_udf", "scala_udf", "java_udf".
```

Comment:
function type may sounds like scalar/aggregate/table function etc.

I don't have a good idea now, maybe `source`? cc @HyukjinKwon

Reply:
I think function type is fine. I see some refs e.g.) https://www.dotnettricks.com/learn/sqlserver/different-types-of-sql-server-functions

Reply:
> function type may sounds like scalar/aggregate/table function etc.

Ohh, yeah

Reply:
okay `source` makes sense